### PR TITLE
Add Prometheus jobs to fetch k8s resource metrics

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
This PR adds three Prometheus jobs to retrieve resources metrics.

**Which issue(s) this PR fixes**:

Ref https://github.com/pipe-cd/pipe/issues/1867

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/pipecd/values.yaml [hunk 1]
```diff
@@ -211,6 +214,56 @@ prometheus:
               action: keep
               regex: admin
 
+        # Scrape config for API servers.
+        - job_name: 'kubernetes-apiservers'
```
Comment (nghialv) [inline manifests/pipecd/values.yaml:218]: nit: Remove `'` to be unified with other jobs.

Comment (nakabonne) [reply manifests/pipecd/values.yaml:218]: for sure

**Comment by nakabonne** [comment]
applied

**Comment by nghialv** [comment]
Nice.
/lgtm

**Comment by khanhtc1202** [comment]
Way to go 🚀 
/approve

**Pull request merged**